haystack/nodes/prompt/prompt_template.py

```diff
@@ -268,9 +268,14 @@ def prepare(self, *args, **kwargs) -> Dict[str, Any]:
                 if param in kwargs:
                     params_dict[param] = kwargs[param]
 
-        if set(params_dict.keys()) != set(self.prompt_params):
+        if not set(self.prompt_params).issubset(set(params_dict.keys())):
             available_params = set(list(params_dict.keys()) + list(set(kwargs.keys())))
```

Comment:
```suggestion
            available_params = set([*params_dict.keys(), *kwargs.keys()])
```

Might as well simplify this too since we're at it.